lib/active_record/connection_adapters/trilogy_adapter.rb

```diff
@@ -195,7 +284,7 @@ def reconnect
 
         def sync_timezone_changes(conn)
           # Sync any changes since connection last established.
-          if default_timezone == :local
+          if ActiveRecord.default_timezone == :local
```

Comment:
Sounds like we're missing a test here; under 7.1, these values can differ

Reply:
To simplify have now just added `#default_timezone` to `TrilogyAdapter` if AR <= 7.0.